bentoml/_internal/runner/batching.py

```diff
@@ -1,71 +0,0 @@
-from __future__ import annotations
-
-import typing as t
-from typing import TYPE_CHECKING
-
-from bentoml.exceptions import InvalidArgument
-
-from .utils import Params
-from .container import AutoContainer
-
-if TYPE_CHECKING:
-    from .container import Payload
-
-
-def batch_params(
-    paramss: t.Sequence[Params[Payload]],
-    batch_dim: int,
-    # TODO: support mapping from arg to batch dimension
-) -> tuple[Params[t.Any], list[int]]:
-    if len(paramss) == 0:
-        return (Params(), [])
-
-    args: list[t.Any] = []
-    kwargs: dict[str, t.Any] = {}
-
-    if len(paramss[0].args) > 0:
-        arg_idxs = iter(range(len(paramss[0].args)))
-
-        idx = next(arg_idxs)
-        batched, first_indices = AutoContainer.from_batch_payloads(
-            [params.args[idx] for params in paramss],
-            batch_dim=batch_dim,
-        )
-        args.append(batched)
```

Comment:
We should prefer
* generators & iterators
* then list comprehension
* then `append` or `expand`
in performance sensitive scenarios